docs/src/v1-api/packaging/nodejs-init-project.md

```diff
@@ -0,0 +1,123 @@
+# initialize nodejs project + dev shell
+
+## load shell with nodejs + npm
+```console tesh-session="next-app" tesh-setup="setup.sh"
+$ nix-shell -p https://dream2nix.dev -A devShells.nodejs
+```
+
+## create my-app
+```console tesh-session="next-app"
+npx create-next-app my-app
+```
+This creates `./my-app/package.json` and more, using `create-next-app` as a helper.
+
+## create `my-app.nix`
+`my-app.nix`
+```nix
+{config, lib, dream2nix, ...}: {
+
+  imports = [
+    # default module to create a nodejs package
+    dream2nix.modules.nodejs.mkDerivation
+    # get package dependencies from package-lock
+    dream2nix.modules.nodejs.package-lock
+  ];
+
+  # Allows to manipulate dependency builds
+  overrides.local.path = ./overrides;
+
+  src = ./my-app;
+
+  # add more mkDerivation attributes here to customize...
+}
+```
+
+## create `my-app-shell.nix` for your dev shell
+`my-app-shell.nix`
+```nix
+{config, lib, dream2nix, ...}: {
+
+  imports = [
+    # the default dev shell for nodejs
+    dream2nix.modules.nodejs.mkShell
+    # adds dependencies of my-app to the dev shell
+    dream2nix.modules.nodejs.package-lock
```

Comment:
Why are these magic imports you have to replicate in each submodule instead of enable options provided by modules that you can just import once at a higher level (probably `default.nix` here)?

Reply:
My intention was to keep evaluation overhead minimal. I wanted to avoid the problem of always having to import all modules, like seen in nixos as it appears to be limiting scalability and slow down evaluation.

I'm not too sure about my judgement here. Maybe it's not worth while and we should switch back to just using `.enable = ...`.

Reply:
I personally like importing the modules, because besides possible evaluation overhead IMO it can make it clearer which options are in scope.

> instead of enable options provided by modules that you can just import once at a higher level (probably default.nix here)

Just to avoid misunderstandings: Those imports also affect all modules which end up the same config (or derivation in the context of drv-parts). So importing them at a higher level would also work. An argument for .enable could be made for the inverse, i think: .enable would allow us to then disable again in specific cases via `enable = False` which isn't as easy for imports.